cm/cm_heat_demand/tools/areas.py

```diff
@@ -0,0 +1,192 @@
+    if n_label > 0:
+        end, start, sorted_array = get_browsing_indexes(
+            labels_array=labels_array,
+            pixel_filtered_map=pixel_filtered_map,
+            n_label=n_label,
+        )
+
+        for i, (start_, end_) in enumerate(zip(start, end)):
+            area = sorted_array[start_:end_, 3]
+            area_potential = np.sum(area)
+            if area_potential >= district_heating_zone_threshold:
+                # i+1 because labeling starts from 1 and not from 0
+                # factor 0.001 for conversion from MWh/ha to GWh/ha
+                areas_potential[i + 1] = np.around(np.sum(area_potential) / 1000, 2)
+
+    areas = areas_potential[labels_array]
+    filtered_map = pixel_filtered_map * (areas > 0).astype(int)
+    total_potential = np.sum(areas_potential)
+    return areas, filtered_map, total_potential, areas_potential[1:]
+
+
+def get_areas(
+    heat_density_map: str,
+    pixel_threshold: float,
+    district_heating_zone_threshold: float,
+):
+    """
+    This function applies a filter on each individual pixel
+    as well as on all the pixels belonging to the same area.
+
+    These areas are defined by the set of adjacent pixels
+    that have passed the first filter.
+
+    Inputs :
+        * heat_density_map :
+            path to the clipped raster (MWh) - a value between 0 and 1.000.
+        * pixel_threshold :
+            threshold that each pixel must reach (MWh) - a value between 0 and 500.
+        * district_heating_zone_threshold :
+            threshold that each zone must reach (GWh/a).
+
+    Outputs :
+        * areas :
+            - map where the value of the pixels belonging to a given area corresponds
+            to the potential of this area.
+        * filtered_map :
+            - map where the values of the pixels not belonging to a filtered area
+            are set to zero.
+        * total_potential :
+            - sum of the potential of each zone.
+        * total_heat_demand :
+            - total heat demand of the map (in GWh)
+        * areas_potential :
+            - list of the potential of each zone.
+
+            The pixels that do not pass the thresholds (and therefore defined
+            with a potential equal to zero) belong to the first zone.
+            For example : [0, "1st area potential", ..., "nth area potential"]
+            - NB: this zone is not interesting, therefore only the potential of
+                the other zones is returned in practice
+    """
+    array_map, geo_transform = read_raster(raster=heat_density_map)
+    # array_map units : MWh
+
+    total_heat_demand = np.around(np.sum(array_map) / 1000, 2)
+    # total_heat_demand units : GWh
+
+    pixel_filtered_map = array_map * (array_map > pixel_threshold)
+    # pixel_filtered_map units : MWh
+
+    district_heating_zone_threshold = district_heating_zone_threshold * 1000
+    # district_heating_zone_threshold units : MWh (after the conversion)
```

Comment:
Why are the comments below the code?

Reply:
no particular reason, they are above the code now

Reply:
Thanks, it's more understandable to choose a way of doing things and stick to it!